test/studies/isx/isx-bucket-spmd.chpl

```diff
@@ -0,0 +1,424 @@
+// buckets to be mapped to each locale, potentially.
+//
+
+//
+// Top priorities:
+// - timer insertion
+// - performance analysis / chplvis
+
+// TODO:
+// * variants that we should try
+//   - multiple buckets per locale w/out any atomic ints
+//   - some sort of hybrid between the current 1 bucket per locale scheme
+//     and the previous?
+
+//
+// TODO: What are the potential performance issues?
+// * put as one message?
+//   - not currently happening due to int(32)/int(64) mismatch that
+//     prevents direct copying -- will fix in future revision to
+//     see performance impact.
+// * how do Ben's locality optimizations do?
+// * does returning arrays cost us anything?  Do we leak them?
+// * other?
+// * (this would be a good chplvis demo)
+//
+
+//
+// We want to use block-distributed arrays (BlockDist) and barrier
+// synchronization (Barrier).
+//
+use BlockDist, Barrier;
+
+//
+// The type of key to use when sorting.
+//
+config type keyType = int(32);
+
+//
+// The following options respectively control...
+// - whether or not to print debug information
+// - whether or not to do a test run (results in small problem sizes)
+// - whether or not to run quietly (squashes successful verification messages)
+// - whether or not to print the execution time configuration
+//
+config const debug = false,
+             testrun = debug,
+             quiet = false,
+             printConfig = !quiet;
+
+
+//
+// Define three scaling modes: strong scaling, weak scaling, and
+// weakISO (in which the width of the buckets is held constant).
+//
+enum scaling {strong, weak, weakISO};
+
+//
+// Which scaling mode should the program be run in?
+//
+config const mode = scaling.weak;
+
+//
+// The number of keys is defined in terms of 'n', though whether this
+// represents the total number of keys or the number of keys per
+// bucket depends on whether we're running in a strong or weak scaling
+// mode.  When debugging, we run a smaller problem size.
+//
+config const n = if testrun then 32 else 2**27;
+
+//
+// TODO: permit these to be configurable
+//
+const bucketsPerLocale = 1;
+const numBuckets = numLocales;
+
+//
+// The total number of keys
+//
+config const totalKeys = if mode == scaling.strong then n
+                                                   else n * numBuckets;
+
+//
+// The number of keys per bucket -- this is approximate for strong
+// scaling if the number of buckets doesn't divide 'n' evenly.
+//
+config const keysPerBucket = if mode == scaling.strong then n/numBuckets
+                                                       else n;
+
+
+//
+// The bucket width to use per bucket when running in weakISO mode
+//
+config const isoBucketWidth = if mode == scaling.weakISO then 8192 else 0;
+
+
+//
+// (Issue a warning if this has been set in modes other than weakISO)
+//
+if !quiet && mode != scaling.weakISO && isoBucketWidth != 0 then
+  warning("Note that isoBucketWidth has no effect for weakISO scaling mode");
```

Comment:
Oh, now I see!  I thought you were saying that the warning wasn't generated when the comment said it was.  It's actually that the warning message is incorrect.  I agree, thanks for catching that.